### DISCUSSION TITLE
Welcome to unchained-engine discussions!

### DISCUSSION THREAD

**Discussion by fliptation** [unchainedshop/unchained#280] [General]
## 👋 Welcome!
  We’re using Discussions as a place to connect with other members of our community. We hope that you:
  * Ask questions you’re wondering about.
  * Share ideas.
  * Engage with other community members.
  * Welcome others and are open-minded. Remember that this is a community we
  build together 💪.

  To get started, comment below with an introduction of yourself and tell us about what you do with this community.

**Comment by shubham-kaushal** [comment]
Hello Unchained Community!

I am starting this discussion to know more about Unchained and the benefits of choosing Unchained over Reaction Commerce (both share the same tech stacks and similar approach). Also, I would love to get an insight into the PIM (inbuilt features of unchained), and using unchained as headless Serverless and with Multi-Tenancy architecture use case.

cc: @pozylon

**Reply by harryadel** [reply]
Hello @shubham-kaushal, Glad to see you asking and starting a discussion, Let's dissect your question into more concrete points:
 
> know more about Unchained and the benefits of choosing Unchained over Reaction Commerce (both share the same tech stacks and similar approach)

If you'd like to learn more about Unchained, you can head over to the [docs](https://docs.unchained.shop/) where you can learn in detail about the architecture or the [developers section](https://developers.unchained.shop/).

What are the benefits to choosing Unchained over Reaction? Well, this totally depends on your needs and what you're trying to achieve. I've not used Reaction Commerce before so I can't comment on it but as for Unchained, if I were to briefly define key offerings of Unchained they'd be:

- [Variety of modules](https://developers.unchained.shop/engine) to manage multiple aspects of your E-commerce store.
- Integrations, whether using our [built in-tools](https://docs.unchained.shop/api/entities/#bulk-import-api) or tools of your own to easily integrate Unchained with your systems/data. 
- [Layered Approach](https://docs.unchained.shop/concepts/architecture/#layered-approach), this allows you to pick and choose the amount of features/depth you'd like to tap into so it doesn't bog your application with needless stuff.  

Also, some small note here, technically speaking Unchained and Reaction share the same heritage (i.e. Meteor). Reaction completely migrated off of it but we're still [on the way](https://github.com/unchainedshop/unchained/issues/115), though don't get me wrong we utilize the same technology (GraphQL, Apollo, MonogDB) it's just the way the engines are powered (Meteor vs Meteor-less).

 > I would love to get an insight into the PIM
 
 > using unchained as headless Serverless and with Multi-Tenancy architecture use case.
 
 As for those two points, I believe @pozylon can offer a better insight since he worked intimately with our PIM system and deployed/managed multiple architectures for our clients.
 
 Cheers, and let us know how we can help you moving forward!

**Reply by pozylon** [reply]
PIM: We are working on building a demo environment so you can try out the Controlpanel to modify products, else just use Akeneo as a PIM and the open-source connector to sync products/catalogs to Unchained
Unchained serverless: Unchained is not built in a way that supports serverless architecture, we might achieve it in the future when we are off meteor but for the moment you have to get a long with a containerized approach
Multi-tenancy: Classic multi-tenance is usually not needed as you can build multiple UI's themed differently pointing to the same backend with the headless approach, you could then add 2 assortments: "Client A" and "Client B" and show the appropriate

**Comment by shubham-kaushal** [comment]
Hello @harryadel and @pozylon, Thanks for sharing insight on the Unchained. We were busy with releasing our open source UI framework https://github.com/numldesign/numl and building Serverless SaaS Boilerplate that will be releasing soon for testing.

Continue to my question of considering Unchained into our Serverless App. I am required few more insights that will help us in making our decisions. We want to use Unchained as Headless commerce for our Multi-Tenant Serverless SaaS product. Our tech stacks are Node.Js, Typescript, React.Js, Next.Js, DynamoDB, Apollo GraphQL, Elasticsearch, S3, Cloudfront, Lambda, Cognito, and API Gateway.

As you will notice we are using DynamoDB by default and Unchained uses MongoDB by default. The good point is we both have taken a layered approach of microservices to build our products. 

1. What are the flexibility and challenges we will be having to adopt Unchained by migrating the database of Unchained from MongoDB to DynamoDB and with Typescript because we heavily rely on Typescript?
2. What's the roadmap of unchained and how much support we will be getting from the unchained team if we adopt Unchained for our commerce module for our Serverless SaaS product as currently there is no community help available for Unchained.

Thanks!

**Reply by schmidsi** [reply]
Hello @shubham-kaushal , sorry for the late reply. We are currently migrating away from Meteor and to a more modular architecture based on Typescript. This will probably make it easier to switch out the database layer. But support for other databases is not our priority at the moment. But if you come up with a good solution that is adding such a functionality, we are happy to review your concept and pull requests and give feedback.

Please note that we are a bootstrapped company and mainly use the Unchained Engine for our own customers. So we try to be available online for questions as much as possible, but sometimes paying customers just have priority.

**Comment by cloudzeus** [comment]
Hello we have tried to install you application in order to evaluate and use it to our company with no luck. I have also send an email asking for your Professional support but I haven't got any answer.

**Reply by pozylon** [reply]
@cloudzeus Oh that's a pity, we do deliver professional support and you could also setup a trial engine so you could test the API. We've seen your mail and will respond until end of week with our rates.
